cpp/src/arrow/flight/sql/column_metadata.h

```diff
@@ -78,6 +81,10 @@ class ColumnMetadata {
   /// \return The table name.
   arrow::Result<std::string> GetTableName() const;
 
+  /// \brief  Return the type name set in the KeyValueMetadata.
```

Comment:
can you expand the comments here what is type name?

Reply:
Ping @iurysalino @jduo would it be possible to clarify things here before merging?

Reply:
```suggestion
  /// \brief  Return the data-source specific name for the data type of the column.
```

Reply:
```suggestion
  /// \brief  Return the data source-specific name for the data type of the column.
```

Reply:
I improved the doc here using James suggestion.

Reply:
Can the clarification be done in the format itself as well?

Reply:
Sure, I already did it